File: sdk/src/test/java/com/cxense/cxensesdk/SdkInterceptorTest.java
package com.cxense.cxensesdk;

import org.junit.Test;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.mockwebserver.MockResponse;
import okhttp3.mockwebserver.RecordedRequest;

import static org.hamcrest.Matchers.allOf;
import static org.hamcrest.Matchers.containsString;
import static org.junit.Assert.assertThat;

/**
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-06-05).
 */
public class SdkInterceptorTest extends BaseInterceptorTest {
    @Test
    public void intercept() throws Exception {
        mockWebServer.enqueue(new MockResponse());

        OkHttpClient okHttpClient = new OkHttpClient()
                .newBuilder()
                .addInterceptor(new SdkInterceptor(BuildConfig.SDK_NAME, BuildConfig.VERSION_NAME))
                .build();
        okHttpClient.newCall(new Request.Builder().url(mockWebServer.url("/")).build()).execute();

        RecordedRequest request = mockWebServer.takeRequest();
        String path = request.getPath();
        assertThat(path, allOf(
                containsString("sdk=" + BuildConfig.SDK_NAME),
                containsString("sdkp=android"),
                containsString("sdkv=" + BuildConfig.VERSION_NAME)
        ));
    }

}

File: sdk/src/test/java/io/piano/android/cxense/SdkInterceptorTest.kt
package io.piano.android.cxense

import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.mockwebserver.MockResponse
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotNull

class SdkInterceptorTest : BaseInterceptorTest() {

    @Test
    fun intercept() {
        mockWebServer.enqueue(MockResponse())
        OkHttpClient.Builder()
            .addInterceptor(SdkInterceptor(SDK_NAME, SDK_VERSION_NAME))
            .build()
            .newCall(
                Request.Builder()
                    .url(mockWebServer.url("/"))
                    .build()
            )
            .execute()
        with(mockWebServer.takeRequest().requestUrl) {
            assertNotNull(this)
            assertEquals(SDK_NAME, queryParameter(SdkInterceptor.ARG_SDK_NAME))
            assertEquals(SDK_VERSION_NAME, queryParameter(SdkInterceptor.ARG_SDK_VERSION))
            assertEquals(SdkInterceptor.VALUE_SDK_PLATFORM, queryParameter(SdkInterceptor.ARG_SDK_PLATFORM))
        }
    }

    companion object {
        private const val SDK_NAME = "name"
        private const val SDK_VERSION_NAME = "version"
    }
}
